tests/ImageSharp.Tests/Formats/Pbm/PbmEncoderTests.cs

```diff
@@ -0,0 +1,144 @@
+                        Assert.Equal(pbmColorType, meta.ColorType);
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [WithFile(BlackAndWhitePlain, PixelTypes.Rgb24)]
+        public void PbmEncoder_P1_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.BlackAndWhite, PbmEncoding.Plain);
+
+        [Theory]
+        [WithFile(BlackAndWhiteBinary, PixelTypes.Rgb24)]
+        public void PbmEncoder_P4_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.BlackAndWhite, PbmEncoding.Binary);
+
+        /* Disabled as Magick throws an error reading the input image
+        [Theory]
+        [WithFile(GrayscalePlain, PixelTypes.Rgb24)]
+        public void PbmEncoder_P2_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.Grayscale, PbmEncoding.Plain);
+        */
+
+        [Theory]
+        [WithFile(GrayscaleBinary, PixelTypes.Rgb24)]
+        public void PbmEncoder_P5_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.Grayscale, PbmEncoding.Binary);
+
+        /* Disabled as Magick throws an error reading the input image
+        [Theory]
+        [WithFile(RgbPlain, PixelTypes.Rgb24)]
+        public void PbmEncoder_P3_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.Rgb, PbmEncoding.Plain);
+        */
+
+        [Theory]
+        [WithFile(RgbBinary, PixelTypes.Rgb24)]
+        public void PbmEncoder_P6_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.Rgb, PbmEncoding.Binary);
+
+        private static void TestPbmEncoderCore<TPixel>(
+            TestImageProvider<TPixel> provider,
+            PbmColorType colorType,
+            PbmEncoding encoding,
+            bool useExactComparer = true,
+            float compareTolerance = 0.01f)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            using (Image<TPixel> image = provider.GetImage())
+            {
+                var encoder = new PbmEncoder { ColorType = colorType, Encoding = encoding };
+
+                using (var memStream = new MemoryStream())
+                {
+                    image.Save(memStream, encoder);
+                    memStream.Position = 0;
+                    using (var encodedImage = (Image<TPixel>)Image.Load(memStream))
+                    {
+                        TgaTestUtils.CompareWithReferenceDecoder(provider, encodedImage, useExactComparer, compareTolerance);
```

Comment:
maybe we should put this method in some common place for testing?

Reply:
Tiff has a very similar utility also. So, now we have three formats using similar code.
Please advise on a location of this utility class and I'll refactor all three of them to a single location.

Reply:
I would put it in `ImageSharp.Tests\TestUtilities\ImageComparison`. Maybe call the class `ImageComparingUtils.cs`?

Reply:
Done this, no code changes required to the TestUtils.